contrib/platform/test/com/sun/jna/platform/win32/Advapi32Test.java

```diff
@@ -1950,12 +1950,14 @@ public void testDecryptFile() throws Exception {
 
         // decrypt a read only file
         file.setWritable(false);
-        assertFalse(Advapi32.INSTANCE.DecryptFile(lpFileName, new DWORD(0)));
-        assertEquals(WinError.ERROR_FILE_READ_ONLY, Kernel32.INSTANCE.GetLastError());
+        boolean successful = Advapi32.INSTANCE.DecryptFile(lpFileName, new DWORD(0));
+        if(! successful) {
```

Comment:
What are the circumstances where a read-only file (`setWritable(false)`) can be decrypted? That seems odd that I can decrypt (write) a file that is not allowed to be written.

Reply:
Actually the `setWriteable` call succeeds (sorry only german screenshots available):

![image](https://github.com/java-native-access/jna/assets/2179736/e03b94a1-8e6c-46c3-9cba-0dca22bd9c2e)

What I think is, that the assumption is wrong, that the "read-only" state of the file matters. In the end windows could just consider the encryption state a meta-data flag:

![image](https://github.com/java-native-access/jna/assets/2179736/f56d1130-6a2e-48a5-a081-fd0593a8bf67)

Windows/NTFS can transparently encrypt the file - and not let userspace modify the content (you are asking the OS to do things, userspace rules don't necessarily apply). Opening the file with notepad++ shows, that I indeed could not modify the file

Reply:
Not for this PR, but would be nice to know reliably what Windows does here and make the test deterministic.